pkg/controller/controlplanecerts/controlplanecerts_controller.go

```diff
@@ -285,9 +286,13 @@ func (r *ReconcileControlPlaneCerts) generateControlPlaneCertsSyncSet(cd *hivev1
 		if cd.Status.APIURL == "" {
 			return nil, errors.New("API URL not set in ClusterDeployment")
 		}
+		apidomain, err := defaultControlPlaneDomain(cd.Status.APIURL)
```

Comment:
The `cd.Status.APIURL` is not necessarily the API URL from the kubeconfig. It could be the override API URL. Although, I think that it is a deficiency that it could be the override API URL. Since https://github.com/openshift/hive/pull/777, the `cd.Status.APIURL` could be set after the clusterdeployment has been set to installed.

Reply:
In any event, I think we want to grab the API URL directly from the kubeconfig rather than relying on the field set in the clusterdeployment status.
```
remoteclient.NewBuilder(...).UseSecondaryAPIURL().APIURL()
```